packages/devtools_app/test/flutter/inspector_screen_test.dart

```diff
@@ -317,4 +339,35 @@ void main() {
   });
 }
 
-class MockInspectorController extends Mock implements InspectorController {}
+class MockInspectorService extends Mock implements InspectorService {}
+
+class MockInspectorTreeController extends Mock implements InspectorTreeController {}
+
+class TestInspectorController extends Mock implements InspectorController {
```

Comment:
Suggestion: implement this as a Fake instead of a mock.

Then, instead of using when() calls in this factory constructor, provide real implementations for those methods.

```dart
class TestInspectorController extends Fake implements InspectorController {
  @override
  InspectorTreeNode get selectedNode => node;
  @override
  void addSelectionListener(VoidCallback listener) => listeners.add(listener);

  // Similarly for the rest.
}
```

Reply:
Lets file bugs on how to test this code better but not do any major work on improving the testing for a couple weeks.

Reply:
very neat! I was googling for better solution last night and didn't know about Fake. I'll look into this.

Reply:
@jacob314  the test for this widget (InspectorScreenDetailsTab) should be easy but the `StoryOfYourFlexWidget` one is hard since I have to fake lot of stuffs (service, group manager). While we are on it, can I temporarily disable the golden test for that? Other than that the tests seem to be working fine.

Reply:
That sounds good. File a bug for the test you have to disable and reference the bug in the code where you skip the test.